memory-profiling.md

```diff
@@ -123,126 +131,112 @@
+- **preferred over snapshot comparisons** to track down memory leaks
+- *blue* bars show memory allocations
+- *grey* bars show memory deallocations
 
-##### Ensuring GC
+#### Allocation Stack
 
-Although as mentioned above before a heapdump is taken all garbage is collected, I found that manually triggering
-garbage collection and forcing the GC to compact yields better results. The reason for this is unclear to me.
+[slides](https://speakerdeck.com/addyosmani/javascript-memory-management-masterclass?slide=102)
+[watch](https://youtu.be/LaxbdIyBkL0?t=51m30s)
 
-Add the below snippet to your code which will only activate if you are exposing the garbage collector:
+![allocation-stack](assets/memory-allocation-stack.png)
 
-```js
-// shortest
-if (typeof gc === 'function') setTimeout(gc, 1000);
+- _Allocation_ view (selectable in top left) shows allocations grouped by function and whose
+  traces can be followed to see the function code responsible for the allocation
 
-// longer in order to see indicators of gc being performed
-if (typeof gc === 'function') {
-  setTimeout(function doGC() { gc(); process.stdout.write(' gc ') }, 1000);
-}
-```
+#### Recording Allocation Timeline with Node.js
 
-Then run your app with the appropriate flags to expose the gc and force compaction.
+- run app via `node --inspect` or `node --inspect-brk`
+- open DevTools anywhere and click on the Node.js icon in the upper left corner to open a
+  dedicated Node.js DevTools instance
+- select the _Memory_ tab and there select _Record allocation timeline_ and then click _Start_
+  - if you launched with `--inspect-brk` go back to the source panel to start debugging and
+    then return to _Memory_ tab
+- stop profiling via the red circle on the upper left and examine the timeline and related
+  snapshots
+- notice that the dropdown on the upper left has an _Allocations_ option which allows you to
+  inspect allocations by function
 
-```
-node --always-compact --expose-gc app.js
-```
+![memory-hog-allocation-timeline](assets/memory-allocation-timeline.gif)
 
-Now you can wait for the garbage collection to occur and take a heapdump right after.
+### DevTools Allocation Profile
 
-Alternatively you can add a hook to your app, i.e. a route, that will trigger manual `gc` and invoke that before taking
-a heapdump.
+[read](https://developers.google.com/web/tools/chrome-devtools/memory-problems/#allocation-profile)
 
-#### Considerations to make code easier to debug
+- helps identify functions responsible for allocating memory
+- in case of memory leaks or performance issues due to lots of allocatd objects it can be used
+  to track down which functions allocate most memory
 
-The usefulness of the information presented in the below views depends on how you authored your code. Here are a few
-rules to make your code more debuggable.
+![memory-hog-allocation-timeline](assets/memory-allocation-timeline.gif)
```

Comment:
Wrong gif?

Reply:
Fixed in a0a098d2ad4e72edd446729a5b469fb2c43b3c33